helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -1078,12 +1081,16 @@ private void updateDisabledInstances(Collection<InstanceConfig> allInstanceConfi
     // Move the calculating disabled instances to refresh
     _disabledInstanceForPartitionMap.clear();
     _disabledInstanceSet.clear();
+    _disabledInstancesForAllPartitionsSet.clear();
     for (InstanceConfig config : allInstanceConfigs) {
       Map<String, List<String>> disabledPartitionMap = config.getDisabledPartitionsMap();
-      if (config.getInstanceOperation().getOperation()
-          .equals(InstanceConstants.InstanceOperation.DISABLE)) {
+      // Treat instance as disabled if it has "DISABLE" operation or "ALL_RESOURCES" in the disabled partition map
+      if (config.getInstanceOperation().getOperation().equals(InstanceConstants.InstanceOperation.DISABLE)) {
         _disabledInstanceSet.add(config.getInstanceName());
       }
+      if (disabledPartitionMap.containsKey(InstanceConstants.ALL_RESOURCES_DISABLED_PARTITION_KEY)) {
+        _disabledInstancesForAllPartitionsSet.add(config.getInstanceName());
```

Comment:
Not make it in base controller. This resource assignment, make it in resource controller provider

Reply:
@junkaixue moved this into the resource controller data provider. 
the `disabledInstancesForAllPartitions` is only maintained int he resource controller provider
`getDisabledInstancesForPartition` is overriden in resource controller data , calls super method then applies the instances from the `disabledInstancesForAllPartitions`